nilearn/plotting/displays/_figures.py

```diff
@@ -104,3 +130,355 @@ def savefig(self, output_file=None):
+                        # this does not share and edge, so try again
+                        continue
+                    for e in segments[current_vertex]:
+                        if e is not None and self._do_segs_intersect(
+                            *xy1, *xy2, *e
+                        ):
+                            # this one crosses boundary, so try again
+                            fail |= True
+                    if fail:
+                        continue
+                    next_index = shortest_idx
+
+                # if none of those five worked, then pick the next nearest and
+                # handle later
+                if next_index == 9999:
+                    next_index = np.argmin(remaining_distances)
+
+            else:
+                next_index = np.argmin(remaining_distances)
+
+            closest_vertex = remaining_vertices[next_index]
+            if remaining_distances[0, next_index] > last_distance * 3:
+                sorted_vertices.append(centroids[prev_first])
+                sorted_vertices.append(np.array([None] * 3))
+                prev_first = closest_vertex
+
+            visited_vertices.add(closest_vertex)
+            sorted_vertices.append(centroids[closest_vertex])
+
+            # Move to the closest vertex and repeat the process
+            current_vertex = closest_vertex
+            last_distance = remaining_distances[0, next_index]
+
+        # at the end we append the first one again to close the outline
+        sorted_vertices.append(centroids[prev_first])
+
+        return np.asarray(sorted_vertices)
+
+    def _get_faces_on_edge(self, parc_idx):
+        """Identify which faces lie on the outeredge of the parcellation \
+        defined by the indices in parc_idx.
+
+        Parameters
+        ----------
+        parc_idx : numpy.ndarray, indices of the vertices
+            of the region to be plotted
+        """
+        # count how many vertices belong to the given parcellation in each face
+        verts_per_face = np.isin(self._faces, parc_idx).sum(axis=1)
+
+        # test if parcellation forms regions
+        if np.all(verts_per_face < 2):
+            raise ValueError("Vertices in parcellation do not form region.")
+
+        vertices_on_edge = np.intersect1d(
+            np.unique(self._faces[verts_per_face == 2]), parc_idx
+        )
+        faces_outside_edge = np.isin(self._faces, vertices_on_edge).sum(axis=1)
+
+        return np.logical_and(faces_outside_edge > 0, verts_per_face < 3)
+
+    @staticmethod
+    def _project_above_face(point, t0, t1, t2, elevation=0.1):
+        u = t1 - t0
+        v = t2 - t0
+        # vector normal to plane
+        n = np.cross(u, v)
+        n /= np.linalg.norm(n)
+        p_ = point - t0
+
+        p_normal = np.dot(p_, n) * n
+        p_tangent = p_ - p_normal
+
+        closest_point = p_tangent + t0
+        return closest_point + elevation * n
+
+    @staticmethod
+    def _do_segs_intersect(
```

Comment:
Can you write a short docstring ?

Reply:
Yes, thanks [bb21ae7](https://github.com/nilearn/nilearn/pull/3949/commits/bb21ae756e6f8fa48133bab7a6d486c33a7170bb)